smoke/test_vpc_vpn: Add template wait & refactor

michaelandersen opened the pull request:
- Added a wait for template download as the test was failing in some environments
- Removed debug timer in setup and teardown
- Increased time.sleep before ssh connection from 10 to 20 seconds
- PEP8
- Changed required_hardware tag to true

michaelandersen commented:
@sanjeevneelarapu please review once build checks are completed

michaelandersen commented:
Tested change on a KVM hypervisor:

Test Remote Access VPN in VPC ... === TestName: test_vpc_remote_access_vpn | Status : SUCCESS ===
ok
Test VPN in VPC ... === TestName: test_vpc_site2site_vpn | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 2 tests in 1084.378s

OK

michaelandersen commented:
executed tests with: 
nosetests --with-marvin --marvin-config=./mct-zone1-kvm1.cfg -a tags=advanced  test_vpc_vpn.py

DaanHoogland commented on test/integration/smoke/test_vpc_vpn.py:
```diff
@@ -570,127 +584,139 @@ def test_vpc_site2site_vpn(self):
-            time.sleep(10)
+            self.assert_(
+                natrule is not None, "Failed to create portforward for vm2")
+            time.sleep(20)
```
Is this the essence of this change , @michaelandersen ? seems like it is postponement of execution. Isn't there a condition to check for instead?

michaelandersen replied:
nope not the essence of this change, that was the waiting for the template download which has been implemented using the template.download method.

the time sleep is to catch anything else that could be slow in certain environments, such as vm deployment.

michaelandersen commented:
changed the required_hardware tag to true and rebased to master and retested:
nosetests --with-marvin --marvin-config=../../../mct-zone1-kvm1.cfg -a tags=advanced,required_hardware=true  test_vpc_vpn.py

Test Remote Access VPN in VPC ... === TestName: test_vpc_remote_access_vpn | Status : SUCCESS ===
ok
Test VPN in VPC ... === TestName: test_vpc_site2site_vpn | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 2 tests in 1056.887s

OK

nosetests --with-marvin --marvin-config=../../../mct-zone1-kvm1.cfg -a tags=advanced,required_hardware=false  test_vpc_vpn.py

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK

remibergsma commented:
LGTM, test runs fine:

```
[root@cs1 MarvinLogs]# less test_vpc_vpn_WK8870/results.txt 
Test Remote Access VPN in VPC ... === TestName: test_vpc_remote_access_vpn | Status : SUCCESS ===
ok
Test VPN in VPC ... === TestName: test_vpc_site2site_vpn | Status : SUCCESS ===
ok

----------------------------------------------------------------------
Ran 2 tests in 1137.908s

OK
```
